xdsl/dialects/builtin.py

```diff
@@ -424,34 +424,67 @@ def verify(self) -> None:
 
 
 @irdl_attr_definition
-class BFloat16Type(ParametrizedAttribute, TypeAttribute):
+class _FloatType(ParametrizedAttribute, TypeAttribute, ABC):
+    name = "float_type"
```

Comment:
why name here? and why _irdl_attr_definition? is it because it doesn't work the same way if only subclasses of ParametrizedAttribute have the annotation?

Reply:
I have a feeling that this is a bit of a half-measure, is it that replacing all the uses of the FloatXType in one go is too big a change? It seems worth discussing on how to get rid of FloatXTypes completely, even if it's as follow-up PRs, it would be good to have a plan before we paint ourselves into a corner.

Reply:
Yeah, I tried that, but not sure how to encode a `BFloat16Type` for example and yes, bigger change.
Not sure if this corners us, currently in MLIR is quite inflexibly limited, so I went with the minimal API change for  now.

Reply:
> why name here? and why _irdl_attr_definition? is it because it doesn't work the same way if only subclasses of ParametrizedAttribute have the annotation?

Good point, oversight from trying different associations